src/napari_graph/interops.py

```diff
@@ -0,0 +1,109 @@
+from typing import Any
+
+import networkx as nx
+import numpy as np
+import pandas as pd
+
+from napari_graph.base_graph import BaseGraph
+from napari_graph.directed_graph import DirectedGraph
+from napari_graph.undirected_graph import UndirectedGraph
+
+
+def from_networkx(graph: nx.Graph) -> BaseGraph:
```

Comment:
If this is not implemented as a classmethod, then it should be called `networkx_to_napari_graph` or similar.

Re circular dependencies, you could avoid that by importing within the method.